pandas/core/internals/blocks.py

```diff
@@ -2300,7 +2309,8 @@ def _try_coerce_result(self, result):
         return result
 
     def should_store(self, value):
-        return issubclass(value.dtype.type, np.timedelta64)
+        return (issubclass(value.dtype.type, np.timedelta64) and
+                not is_extension_array_dtype(value))
```

Comment:
maybe we should have ``is_extension_array_scalar_dtype(value)`` to make this more specific?

Reply:
What that would look like for 3rd party extension arrays? I suppose we could look through the `.type` of all the types registered with pandas, but registering is optional...

Reply:
use registering is optional, but doesn't matter here, my point is we dont' have a way of detecting scalar extension types.